oneOfWeighted.js

```diff
@@ -11,20 +11,14 @@ function oneOfWeighted(a, b) {
 
 function oneOfWeightedMain(input, samples) {
   samples = parseSamples(samples)
-  var id = hash2(input, 'oneOfWeighted')
-  var n = samples.length
-  var pRemaining = 1
-  var i = -1
-  var sample
-  var p
-
-  while (++i < n) {
-    sample = samples[i]
-    p = sample[0] / pRemaining
-    pRemaining -= p
-
-    if (flip(id, p)) {
-      return resolve(id, sample[1])
+  var id = hash(input, 'oneOfWeighted')
+  const prob = (id % 1_000_000) / 1_000_000
```

Comment:
`hash()` gives us back a 53-bit int, so I think we might be truncating the range with the `/ 1_000_000`, which could affect the quality of the results. Could we do `id / Number.MAX_SAFE_INTEGER` here instead?

Reply:
It turns out my suggestion fails the tests (the diff threshold is greater than 6.5%), whereas yours doesn't. I can't understand why. So lets go for your change - scratch my suggestion :) Can you help me understand why 1M works better?

```suggestion
  var prob = (id % 1000000) / 1000000
```

Reply:
I tried today again many different settings and don't understand it at all. I figured in the end it doesn't really matter because only the probabilities the user defined in the array are important. It's unlikely they would define values like `0.000004` and want to run the function more than a million times. Right now, somehow 1 million gave the best results with many different things I tried.

On the other hand: If we break backwards compatibility we should do it right the first time. So I'll take some more time to look into alternatives.

Reply:
Thanks @ctholho. I think it might actually matter though - for some reason using 1M  to convert the id number hash to a decimal to results in `oneOfWeighted()` working much more accurately compared to using 2^52 instead. Accurately as in, the probability of a value getting chosen is much closer to the probability defined by the user (based on these tests, it stays within 6.5% of the probability value the user defined).

So even if there's still something for us to figure out, from an empirical point of view, the 1M value you have here is a better choice, so I'll merge it like this!